modules/ting_das/ting_das.module

```diff
@@ -0,0 +1,378 @@
+    ),
+  );
+
+  global $user;
+  $provider = _ding_provider_get_provider('user');
+  $profile = ding_user_provider_profile($user);
+  if ($profile) {
+    $profile = entity_metadata_wrapper('profile2', $profile);
+  }
+
+  $texts = variable_get('ting_das_text', array());
+  $creds = ding_user_get_creds();
+
+  if (isset($creds['resident']) && $creds['resident']) {
+    $form['title'] = array(
+      '#markup' => '<div class="popupbar-title"><strong>' . t('Fill out the form to order at digital copy') . '</strong></div>',
+    );
+
+    $form['description'] = array(
+      '#prefix' => '<div class="form-item form-type-markup description">',
+      '#markup' => check_markup($texts['description']['value'], $texts['description']['format']),
+      '#suffix' => '</div>',
+    );
+
+    $form['entity'] = array(
+      '#tree' => TRUE,
+    );
+
+    $form['entity']['pid'] = array(
+      '#type' => 'value',
+      '#value' => $entity->getId(),
+    );
+
+    $form['entity']['title'] = array(
+      '#type' => 'value',
+      '#value' => $entity->getTitle(),
+    );
+
+    $form['loaner_id'] = array(
+      '#type' => 'value',
+      '#value' => $creds['library_card_number'],
+    );
+
+    $form['name'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Name'),
+      '#default_value' => $profile ? $profile->{'field_' . $provider['module'] . '_name'}->value() : '',
+      '#required' => TRUE,
+    );
+
+    $form['mail'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Mail address'),
+      '#default_value' => $profile ? $profile->{'field_' . $provider['module'] . '_mail'}->value() : '',
+      '#element_validate' => array(
+        'ting_das_mail_validate',
+      ),
+      '#required' => TRUE,
+    );
+
+    $function = $provider['module'] . '_user_branches';
+    $preferred_branch = $profile ? $profile->{'field_' . $provider['module'] . '_preferred_branch'}->value() : '';
+    $form += ding_reservation_default_options_branch('fbs_preferred_branch', $preferred_branch, $function($user));
+
+    $default_interest_period = ding_provider_invoke('reservation', 'default_interest_period');
+    $form += ding_reservation_interest_period_selector('interest_period', $default_interest_period, _ting_das_get_interest_periods());
+
+    if (isset($texts['gdpr'])) {
+      $form['gdpr'] = array(
+        '#prefix' => '<div class="form-item form-type-markup gdpr-text">',
+        '#markup' => check_markup($texts['gdpr']['value'], $texts['gdpr']['format']),
+        '#suffix' => '</div>',
+        '#weight' => isset($texts['placement']) ? $texts['placement'] : 10,
+      );
+    }
+
+    $form['submit'] = array(
+      '#type' => 'submit',
+      '#submit' => array('ting_das_order_form_submit'),
+      '#ajax' => array(
+        'callback' => 'ting_das_order_form_ajax_callback',
+      ),
+      '#value' => t('Order digital copy'),
+      '#attributes' => array(
+        'onclick' => 'wt.sendinfo({ linkId: "order_digital_copy" });',
```

Comment:
Again: Note the dependency on `ding_webtreds`.

Reply:
Is not added to the info file.